Build.ps1

```diff
@@ -0,0 +1,129 @@
+Param(
+    # Archive containing the Ceph Windows binaries, will be fetched using scp.
+    # Can be a local path, a UNC path or a remote scp path.
+    [Parameter(Mandatory=$true)]
+    [string]$CephZipPath,
+
+    # The thumbprint of the X509 certificate used for signing the WNBD driver
+    # and the MSI installer
+    [string]$SignX509Thumbprint = $null
+)
+
+$ErrorActionPreference = "Stop"
+
+function SetVCVars($version="2019", $platform="x86_amd64") {
+    pushd "$ENV:ProgramFiles (x86)\Microsoft Visual Studio\$version\Community\VC\Auxiliary\Build"
+    try {
+        cmd /c "vcvarsall.bat $platform & set" |
+        foreach {
+          if ($_ -match "=") {
+            $v = $_.split("="); set-item -force -path "ENV:\$($v[0])"  -value "$($v[1])"
+          }
+        }
+    }
+    finally {
+        popd
+    }
+}
+
+function Expand7z($archive, $outputDir = ".") {
+    pushd .
+    try {
+        cd $outputDir
+        &7z.exe x -y $archive
+        if ($LastExitCode) { throw "7z.exe failed on archive: $archive"}
+    }
+    finally {
+        popd
+    }
+}
+
+function Sign($x509thumbprint, $path) {
+    & signtool.exe sign /sha1 $x509thumbprint /td SHA256 /v $path
+    if($LASTEXITCODE) { throw "signtool failed" }
+}
+
+function BuildWnbd() {
+
+    pushd $depsBuildDir
+    & git.exe clone https://github.com/cloudbase/wnbd
+    if($LASTEXITCODE) {
+        throw "git failed"
+    }
+    cd wnbd
+
+    & msbuild.exe vstudio\wnbd.sln /p:Configuration=Release
+    if($LASTEXITCODE) {
+        throw "msbuild failed"
+    }
+
+    if($SignX509Thumbprint) {
+        Sign $SignX509Thumbprint .\vstudio\x64\Release\driver\wnbd.sys
+        Sign $SignX509Thumbprint .\vstudio\x64\Release\driver\wnbd.cat
+    }
+
+    copy vstudio\x64\Release\driver\* ..\..\Driver\
+    copy vstudio\x64\Release\libwnbd.dll ..\..\Binaries\
+
+    popd
+}
+
+function GetCephBinaries() {
+    pushd $depsBuildDir
+
+    & scp.exe $CephZipPath ceph.zip
+    if($LASTEXITCODE) {
+        throw "scp failed"
+    }
+
+    if (Test-Path ceph) {
+        rm -Recurse -Force ceph\
+    }
+
+    Expand7z ceph.zip
+    rm ceph.zip
+    cd ceph\
+
+    copy -Path *.dll,`
+    ceph-authtool.exe,`
+    ceph-conf.exe,`
+    ceph-immutable-object-cache.exe,`
+    ceph_radosacl.exe,`
+    ceph_scratchtool.exe,`
+    ceph_scratchtoolpp.exe,`
+    rados.exe,`
+    rbd-mirror.exe,`
+    rbd-wnbd.exe,`
+    rbd.exe,`
+    ceph-dokan.exe `
+    -Destination ..\..\Binaries\
+
+    copy -Path ceph,ceph-coverage,ceph-crash,ceph-debugpack,ceph-post-file,init-ceph -Destination ..\..\Binaries\

```

Comment:
Those Python scripts are currently unsupported. On the other hand, I'm still a bit concerned about hand picking individual files. If any of them will be renamed or removed, this script will get broken.

Reply:
This is the content of the zip file, built with SKIP_TESTS=1, it's full of irrelevant files, adding all of them is definitely a no go:
http://paste.openstack.org/show/ZulNom8oRsvWu3hBAsdF/

Any binary included must be something meaningful for the user.

Reply:
Those ``*test*`` binaries come from a previous job that probably didn't use ``SKIP_TESTS=1``:

```
4/28/2020   3:18 AM        1574400 ceph_test_rados_api_lock_pp.exe
vs
9/21/2020   7:54 PM         203264 ceph-conf.exe
```

Reply:
I found the issue: the Ceph win32_build.sh does not clear the zip before adding new files and CLEAN_BUILD is not set, resulting in content filing up in the zip across build. Fixed here: https://github.com/petrutlucian94/ceph/pull/8

This means that we don't have to filter the content to add here, which is great

Reply:
fwiw, petrutlucian94/ceph#8 only helps if you're using a ``ZIP_DEST`` path other than the default and you're passing ``CLEAN_BUILD``. If you're using the default zip location, ``CLEAN_BUILD`` will clean it up. If you're not passing ``CLEAN_BUILD``, previously generated binaries will be included anyway (e.g. test files from another run that had different flags).

Reply:
I've updated the ceph build script to remove the binaries dir whenever doing a new build, unless ``SKIP_BINDIR_CLEAN`` is set. That will avoid getting binaries from previous builds, that may have used different flags.